server/plugins/schema-plugin-ob-oracle/src/main/java/com/oceanbase/odc/plugin/schema/oboracle/parser/OBOracleGetDBTableByParser.java

```diff
@@ -184,19 +184,22 @@ public List<DBTableConstraint> listConstraints() {
                 constraint.setOrdinalPosition(i.get());
                 constraint.setTableName(removeIdentifiers(createTableStmt.getTableName()));
                 constraint.setDeferability(DBConstraintDeferability.NOT_DEFERRABLE);
-                OutOfLineConstraint ofLineConstraint = (OutOfLineConstraint) element;
-                constraint.setName(removeIdentifiers(ofLineConstraint.getConstraintName()));
+                OutOfLineConstraint outOfLineConstraint = (OutOfLineConstraint) element;
+                if (Objects.nonNull(outOfLineConstraint.getConstraintName())) {
+                    constraint.setName(removeIdentifiers(outOfLineConstraint.getConstraintName()));
+                }
                 constraint.setEnabled(
-                        ofLineConstraint.getState() == null || ofLineConstraint.getState().getEnable() == null
-                                || ofLineConstraint.getState().getEnable());
+                        outOfLineConstraint.getState() == null || outOfLineConstraint.getState().getEnable() == null
+                                || outOfLineConstraint.getState().getEnable());
                 constraint.setValidate(
-                        ofLineConstraint.getState() == null || ofLineConstraint.getState().getValidate() == null
-                                || ofLineConstraint.getState().getValidate());
-                if (ofLineConstraint instanceof OutOfLineCheckConstraint) {
+                        outOfLineConstraint.getState() == null || outOfLineConstraint.getState().getValidate() == null
```

Comment:
leak of setColumnNames when check constraints exists?

Reply:
check constraints do not need to setColumnNames ，the front end only displays `checkClause`